### PR TITLE
Closes #493: Add background gradient class and new spacing option

### LINKED ISSUE #493: Add gradient-title class
## Motivation
I want to enhance my titles - connected to card updates.

## Proposed Resolution
Add new class and documentation
```
.gradient-title {
    background-image: linear-gradient(180deg, transparent, black);
    padding: 8rem 1.25rem 1.25rem 1.25rem;
    color: #ffffff;
}
```

### Describe alternatives you've considered
Padding of 8 rem is achievable with pt-10 class, but we do not have spacing classes to match 1.25rem which is the set padding of cards.

## Additional context
Blocks https://github.com/az-digital/az_quickstart/issues/1672

### PR DESCRIPTION
**Pull request by danahertzberg**
Closes #493

Changes made per discussion on issue:
- New `card` spacing option is added 
- New `bg-gradient-*` class is added
- Documentation is updated

### CODE REVIEW

**Comment by danahertzberg** [comment]
Spacing documentation (working): https://review.digital.arizona.edu/arizona-bootstrap/issue/493/docs/2.0/utilities/spacing/#notation

Gradient documentation (not working): https://review.digital.arizona.edu/arizona-bootstrap/issue/493/docs/2.0/utilities/colors/#background-gradient

**Comment by danahertzberg** [comment]
Review link: https://review.digital.arizona.edu/arizona-bootstrap/issue/493/docs/2.0/utilities/colors/#background-gradient

File: site/content/docs/2.0/utilities/colors.md [hunk 1]
```diff
@@ -85,7 +85,12 @@ Similar to the contextual text color classes, easily set the background of an el
 {{< /example >}}
 
 ## Background Gradient
+Use the `bg-gradient-*` class to apply a gradient background from a color on the bottom to transparent on the top.
+{{< example >}}
```
Comment (joeparsons) [inline site/content/docs/2.0/utilities/colors.md:89]: The failing PR check is coming from linkinator:
```
> linkinator _site --recurse --skip "^(?!http://localhost)" --verbosity error

"file:/arizona-bootstrap-src/_site/docs/2.0/utilities/colors/index.html":521.438-521.441: error: No "p" element in scope but a "p" end tag seen.
```

This is what the rendered HTML looks on the review site (note the `<p></p>` getting wrapped around the example markup:
```


<h2 id="background-gradient">Background Gradient</h2>
<p>Use the <code>bg-gradient-*</code> class to apply a gradient background from a color on the bottom to transparent on the top.
  <div class="bd-example">
    <h3 class="bold p-card pt-8 bg-gradient-black text-white">.bg-gradient-black</h3>
  </div>
  <div class="highlight">
    <pre tabindex="0" class="chroma">
      <code class="language-html" data-lang="html">
        <span class="p">&lt;</span><span class="nt">h3</span> <span class="na">class</span><span class="o">=</span><span class="s">&#34;bold p-card pt-8 bg-gradient-black text-white&#34;</span><span class="p">&gt;</span>.bg-gradient-black<span class="p">&lt;/</span><span class="nt">h3</span><span class="p">&gt;</span>
      </code>
    </pre>
  </div>
</p>

```


```suggestion

{{< example >}}
```

Comment (danahertzberg) [reply site/content/docs/2.0/utilities/colors.md:89]: Oh dang, thanks @joeparsons!

File: scss/_custom-variables.scss [hunk 2]
```diff
@@ -100,6 +100,18 @@ $colors-transparent: map-merge(
 );
 
 
+// >> Gradient Colors
+$gradient-black: linear-gradient(180deg, transparent, $black);
+
+$colors-gradient: ();
+$colors-gradient: map-merge(
+  (
+    "gradient-black":   $black,
```
Comment (joeparsons) [inline scss/_custom-variables.scss:109]: The linting checks are now failing on the `fusv` (find unused sass variables) step:
```
Looking for unused variables
Searching for unused variables in "/arizona-bootstrap-src/site/assets/scss" folder, scss files...
10 total variables.
No unused variables found in "/arizona-bootstrap-src/site/assets/scss!"
Searching for unused variables in "/arizona-bootstrap-src/scss" folder, scss files...
145 total variables.
1 is not used!
Variable $gradient-black is not being used! /arizona-bootstrap-src/scss/_custom-variables.scss:104
Error: Found 1 unused variable in "/arizona-bootstrap-src/scss" folder
```

```suggestion
    "gradient-black": $gradient-black,
```

**Review by mmunro-ltrr — APPROVED** [review]

**Review by joeparsons — APPROVED** [review]

**Review by cpoff — APPROVED** [review]

**Review by trackleft — APPROVED** [review]

**Pull request merged**